### PR TITLE
Allow string item id for Uuids to support classes

### LINKED ISSUE #14165: [Bug]: Exception during classes-rebuild because of types in uuid
### Expected behavior

`bin/console pimcore:deployment:classes-rebuild` executes without any errors

### Actual behavior

The command fails with an exception in UUID.php

```bash
In UUID.php line 96:

  Pimcore\Bundle\UuidBundle\Model\Tool\UUID::setItemId(): Argument #1 ($id) m
  ust be of type int, string given, called in /var/www/pimcore/vendor/pimcore
  /pimcore/bundles/UuidBundle/src/Model/Tool/UUID.php on line 157


pimcore:deployment:classes-rebuild [-c|--create-classes] [-d|--delete-classes]
In UUID.php line 96:

  Pimcore\Bundle\UuidBundle\Model\Tool\UUID::setItemId(): Argument #1 ($id) m
  ust be of type int, string given, called in /var/www/pimcore/vendor/pimcore
  /pimcore/bundles/UuidBundle/src/Model/Tool/UUID.php on line 157
```

### Steps to reproduce

1. Configure the UUID Bundle
2. Create classes with strings as an identifier
3. Run the classes-rebuild command

### Error
The error is because the UUIDListener listens on [DataObjectClassDefinitionEvents::POST_ADD](https://github.com/pimcore/pimcore/blob/11.x/bundles/UuidBundle/src/EventListener/UUIDListener.php#L45) and calls the same function that is called if an asset, document or object is created. The problem is now, that with the [parameter and return types added](https://github.com/pimcore/pimcore/commit/83142b53cfd5a791e3c6167470a66336fd03da22#diff-c5ca413fbc56c00a39bded05525fb3ef63918f07474b3d57298e6fba24634055) the [itemId in the UUID Class](https://github.com/pimcore/pimcore/blob/11.x/bundles/UuidBundle/src/Model/Tool/UUID.php) is type hinted with int. Classes can have and in our case or the demo, always have, a string as an identifier.

### Solution
I don't know why a UUID is created in the first place for the classes. This doesn't make sense in my opinion. So there are two options in my opinion:
1. Allow string as itemId
4. Remove the DataObjectClassDefinitionEvents::POST_ADD listener

I'm happy to create a PR for that if you can tell me which option should be used.

### PR DESCRIPTION
**Pull request by dvesh3**
## Changes in this pull request  
Resolves #14165

### CODE REVIEW

File: doc/Development_Documentation/19_Development_Tools_and_Details/19_UUID_Support.md [hunk 1]
```diff
@@ -11,13 +11,13 @@ pimcore_uuid:
   instance_identifier: 'your_unique_instance_identifier'
 ```
 
-Once set, Pimcore automatically creates an UUID for each newly created document, asset and object. 
+Once set, Pimcore automatically creates an UUID for each newly created document, asset, classes and object.
```
Comment (jdreesen) [inline doc/Development_Documentation/19_Development_Tools_and_Details/19_UUID_Support.md]:
```suggestion
Once set, Pimcore automatically creates an UUID for each newly created document, asset, class and object. 
```

What's the difference between "class" and "object"?

File: bundles/InstallBundle/dump/install.sql [hunk 2]
```diff
@@ -691,7 +691,7 @@ CREATE TABLE `users_workspaces_object` (
 DROP TABLE IF EXISTS `uuids`;
 CREATE TABLE `uuids` (
   `uuid` CHAR(36) NOT NULL,
-  `itemId` int(11) unsigned NOT NULL,
+  `itemId` VARCHAR(50) NOT NULL,
```
Comment (blankse) [inline bundles/InstallBundle/dump/install.sql]: Should be also changed in https://github.com/pimcore/pimcore/blob/11.x/bundles/UuidBundle/src/Resources/install/install.sql
Why is the table also in the InstallBundle? Should be only in UuidBundle, right?

File: bundles/CoreBundle/src/Migrations/Version20230203160742.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20230203160742 extends AbstractMigration
```
Comment (blankse) [inline bundles/CoreBundle/src/Migrations/Version20230203160742.php]: Shouldn't it be a migration of the UuidBundle?

File: doc/Development_Documentation/19_Development_Tools_and_Details/19_UUID_Support.md (hunk 1 above)
Comment (blankse) [reply doc/Development_Documentation/19_Development_Tools_and_Details/19_UUID_Support.md]: "class" is the ClassDefinition and "object" is a DataObject.

File: bundles/InstallBundle/dump/install.sql (hunk 2 above)
Comment (dvesh3) [reply bundles/InstallBundle/dump/install.sql]: Good catch! yes, that should only be in UuidBundle installer.

File: bundles/CoreBundle/src/Migrations/Version20230203160742.php (hunk 3 above)
Comment (dvesh3) [reply bundles/CoreBundle/src/Migrations/Version20230203160742.php]: Moved.

File: bundles/UuidBundle/src/Resources/install/install.sql [hunk 4]
```diff
@@ -1,7 +1,7 @@
 DROP TABLE IF EXISTS `uuids`;
 CREATE TABLE `uuids` (
     `uuid` CHAR(36) NOT NULL,
-    `itemId` int(11) unsigned NOT NULL,
+    `itemId` VARCHAR(50) NOT NULL,
```
Comment (mattamon) [inline bundles/UuidBundle/src/Resources/install/install.sql:4]: How did you determine how long the varchar can be?

Comment (dvesh3) [reply bundles/UuidBundle/src/Resources/install/install.sql:4]: Based on the class definition ID column type..

Comment (dvesh3) [reply bundles/UuidBundle/src/Resources/install/install.sql:4]: please see https://github.com/pimcore/pimcore/blob/f33ee7e25c596201668109648823c41a6d5dd127/bundles/InstallBundle/dump/install.sql#L77

**Review by mattamon — APPROVED** [review]
LGTM

**Pull request merged**